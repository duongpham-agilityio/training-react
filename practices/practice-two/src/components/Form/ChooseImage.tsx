import { memo } from 'react';

// Components
import { Button } from 'components/commons';

// Assets
import UploadIcon from 'assets/icons/upload.svg';

const ChooseImage = ({
  name,
  chooseHandler,
}: {
  name: string;
  chooseHandler: () => void;
}) =>
  !name ? (
    <Button
      label="Upload"
      width="w-lg"
      border="b-lg"
      variant="primary"
      leftIcon={UploadIcon}
      onClick={chooseHandler}
      data-testid="upload-btn"
    />
  ) : (
    <p onClick={chooseHandler}>{name}</p>
  );

export default memo(ChooseImage);
