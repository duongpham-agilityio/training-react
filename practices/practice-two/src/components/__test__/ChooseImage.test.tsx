import { fireEvent, render } from '@testing-library/react';
import { act } from 'react-dom/test-utils';

// Mocks
import './mocks/image.test';

// Component
import ChooseImage from 'components/Form/ChooseImage';

const setup = (props: { name: string; chooseHandler: () => void }) =>
  render(<ChooseImage {...props} />);

describe('ChooseImage component', () => {
  it('Match snapshot', () => {
    const { container } = setup({
      name: 'image.jpg',
      chooseHandler: jest.fn(),
    });

    expect(container).toMatchSnapshot();
  });

  it('Render with props', () => {
    const onClick = jest.fn();
    const { getByRole } = setup({
      name: '',
      chooseHandler: onClick,
    });
    const button = getByRole('button');

    act(() => {
      fireEvent.click(button);
    });

    expect(onClick).toBeCalled();
  });
});
