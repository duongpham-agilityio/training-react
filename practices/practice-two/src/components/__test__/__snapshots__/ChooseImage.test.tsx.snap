// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChooseImage component Match snapshot 1`] = `
<div>
  <p>
    image.jpg
  </p>
</div>
`;
